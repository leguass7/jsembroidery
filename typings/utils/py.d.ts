export function bytearray(bytable: any): Uint8Array;
export function isStr(value: any): boolean;
export function len(value: any): any;
export function isInstance(value: any, Instance: any): boolean;
//# sourceMappingURL=py.d.ts.map