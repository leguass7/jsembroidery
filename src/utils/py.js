/* eslint-disable no-throw-literal */
export function bytearray(bytable) {
  if (bytable === undefined) return new Uint8Array(0);

  const aType = typeof bytable;
  if (aType === 'number') return new Uint8Array(bytable);

  if (aType === 'string') {
    const size = bytable.length;
    const aBytes = new Uint8Array(size);
    for (let i = 0; i < size; i++) {
      aBytes[i] = bytable.charCodeAt(i);
    }
    return aBytes;
  }

  throw 'Error: py.js bytearray';
}

export function isStr(value) {
  return !!(typeof value === 'string');
}

export function len(value) {
  return value ? value.length : 0;
}

export function isInstance(value, Instance) {
  return !!(value instanceof Instance);
}
