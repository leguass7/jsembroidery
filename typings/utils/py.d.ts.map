{"version":3,"file":"py.d.ts","sourceRoot":"","sources":["../../src/utils/py.js"],"names":[],"mappings":"AACA,oDAgBC;AAED,2CAEC;AAED,qCAEC;AAED,+DAEC"}