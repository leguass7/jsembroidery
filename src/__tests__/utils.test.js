import { isStr, isInstance, len, bytearray } from '../utils/py';

class Test {
  constructor() {
    return this;
  }
}

describe('py functions', () => {
  it('isStr', () => {
    expect(isStr('teste')).toBe(true);
  });

  it('isInstance', () => {
    const T = new Test();
    expect(isInstance(T, Test)).toBe(true);
  });

  it('len', () => {
    const arr = [1, 2, 3, 4, 5];
    expect(len(arr)).toBe(5);
    expect(len('value')).toBe(5);
  });

  it('bytearray', () => {
    const arr = bytearray('teste');
    expect(JSON.stringify(arr)).toMatchSnapshot();
  });
});
