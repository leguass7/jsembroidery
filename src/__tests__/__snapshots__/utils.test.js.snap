// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`py functions bytearray 1`] = `"{\\"0\\":116,\\"1\\":101,\\"2\\":115,\\"3\\":116,\\"4\\":101}"`;
